pkg/flare/flare.go

```diff
@@ -0,0 +1,435 @@
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+
+		f, err := w.Create(relPath)
+		if err != nil {
+			return err
+		}
+
+		_, err = io.Copy(f, file)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+	err = filepath.Walk(inputPath, walker)
+	return err
+}
+
+// HasLogFileOutput checks the outputs property under the logging property of the config
+// to make sure logs are configured to be written to file.
+func (f *Flare) HasLogFileOutput(outputs []string) bool {
+	return slices.ContainsFunc(outputs, func(e string) bool {
+		return e != "" && e != "-" && e != "="
+	})
+}
+
+// SupportedLogFormat checks that the string provided is one of the supported LogFormats
+func SupportedLogFormat(format string) bool {
+	v := LogFormat(format)
+	return v == LogFormatJSON || v == LogFormatPlainText
+}
+
+// ProcessLogFiles processes log files in inputPath. If the inputPath is a file, it processes the file.
+// If inputPath is a directory, all files in the directory will be individually processed.
+// Processed files are read line-by-line. Each line is filtered according to the date range,
+// sanitized from secrets, and is written to the file in outputPath.
+func (f *Flare) ProcessLogFiles(inputPath, outputPath string, startDate, endDate *time.Time) (retErr error) {
+	isDir, err := isDirectory(inputPath)
+	if err != nil {
+		return err
+	}
+
+	if !isDir {
+		return f.processLogFile(inputPath, outputPath, startDate, endDate)
+	} else {
+		files, err := os.ReadDir(inputPath)
+		if err != nil {
+			return err
+		}
+
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			err = f.processLogFile(file.Name(), outputPath, startDate, endDate)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func extractDateFromPlainTextLine(line, logDateLayout string) (time.Time, error) {
+	idxs := plainTextLogDateRegex.FindIndex([]byte(line))
+	if idxs == nil {
+		return time.Time{}, ErrDateNotFound
+	}
+
+	strDate := line[idxs[0]+1 : idxs[1]-1]
+	date, err := time.Parse(logDateLayout, strDate)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return date, nil
+}
+
+func extractDateFromJSONLine(line, logDateLayout string) (time.Time, error) {
+	logPropsMap := make(map[string](interface{}))
+	err := json.Unmarshal([]byte(line), &logPropsMap)
+	if err != nil {
+		return time.Time{}, err
+	}
+	lineDateStr, ok := logPropsMap["time"].(string)
+	if !ok {
+		return time.Time{}, ErrExtractDateFromJSONLogLine
+	}
+	lineDate, err := time.Parse(logDateLayout, lineDateStr)
+	if err != nil {
+		return time.Time{}, err
+	}
+	return lineDate, nil
+}
+
+func (f *Flare) handleLogLine(line string, start, end *time.Time) (string, error) {
+	lineDate, err := dateExpractors[f.logFormat](line, f.LogDateLayout)
+	if err != nil {
+		return "", err
+	}
+	if (start == nil || lineDate.Equal(*start) || lineDate.After(*start)) &&
+		(end == nil || lineDate.Equal(*end) || lineDate.Before(*end)) {
```

Comment:
Please prefer an early-return style, aka avoid indentation.  I believe that it is also clearer:
```suggestion
	if !((start == nil || lineDate.Equal(*start) || lineDate.After(*start)) &&
		(end == nil || lineDate.Equal(*end) || lineDate.Before(*end))) {
		return "", nil
	}
```

Reply:
Done.